include/nbl/video/IPhysicalDevice.h

```diff
@@ -512,11 +512,42 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
             }
         }
 
+        
+        static inline uint32_t getMaxInvocationsPerComputeUnitsFromDriverID(E_DRIVER_ID driverID)
+        {
+            const bool isIntelGPU = (driverID == E_DRIVER_ID::EDI_INTEL_OPEN_SOURCE_MESA || driverID == E_DRIVER_ID::EDI_INTEL_PROPRIETARY_WINDOWS);
+            const bool isAMDGPU = (driverID == E_DRIVER_ID::EDI_AMD_OPEN_SOURCE || driverID == E_DRIVER_ID::EDI_AMD_PROPRIETARY);
+            const bool isNVIDIAGPU = (driverID == E_DRIVER_ID::EDI_NVIDIA_PROPRIETARY);
+            if (isNVIDIAGPU)
+                return 32u * 48u; // RTX 3090 (32 Threads/Warp * 48 Warp/SM)
+            else if (isAMDGPU)
+                return 64u * 1024u; // RX 6900XT (64 Threads/Wave * 1024 Threads/Wave) https://gpuopen.com/learn/optimizing-gpu-occupancy-resource-usage-large-thread-groups/
+            else if (isIntelGPU)
+                return 8u; // Iris Xe HPG (DG2) (8 Threads / XVE) https://www.intel.com/content/www/us/en/develop/documentation/oneapi-gpu-optimization-guide/top/xe-arch.html
```

Comment:
8 is the SIMD unit size

> Each EU is simultaneously multithreaded (SMT) with seven threads. The primary computation unit consists of a 8-wide Single Instruction Multiple Data (SIMD) Arithmetic Logic Units (ALU) supporting SIMD8 FP/INT operations and a 2-wide SIMD ALU supporting SIMD2 extended math operations. Each hardware thread has 128 general-purpose registers (GRF) of 32B wide.

> The SLM is a 128KB High Bandwidth Memory (HBM) accessible from the EUs in the subslice. One important usage of SLM is to share atomic data and signals among the concurrent work-items executing in a subslice. For this reason, if a kernel’s work-group contains synchronization operations, all work-items of the work-group must be allocated to a single subslice so that they have shared access to the same 128KB SLM. 

Its the Subslice that's the equivalent of a CU/SM.

Anyway this is the page you should pay attention to:
https://www.intel.com/content/www/us/en/develop/documentation/oneapi-gpu-optimization-guide/top/thread-mapping.html